DSC/dsc.py

```diff
@@ -471,13 +481,15 @@ def get_host_base_from_uri(blob_uri):
     return netloc[netloc.find('.'):]
 
 def download_external_file(file_uri, download_dir):
+    waagent.AddExtensionEvent(name=ExtensionShortName, op="EnableInProgress", isSuccess=True, message="Downloading from external file")
```

Comment:
Instead of adding events at so many places - why dont you throw error (basically bubble it up) to a common place and then throw an event. That way you dont have to make changes everywhere should you add another param to the event or change existing ones. Something like this:

Try
{
  make sub calls....
}Catch{
 //catch all errors
addextensionevent....
}